File: auto_agent_commit/commit.py
import sys
import json
import os
import subprocess

def execute_command(cmd: list[str]) -> str:
    result = subprocess.run(cmd, capture_output=True, text=True, check=True)
    return result.stdout.strip()

def main():
    try:
        raw_input = sys.stdin.read()
        payload = json.loads(raw_input) if raw_input else {}
        
        workspaces = payload.get("workspacePaths", [os.getcwd()])
        os.chdir(workspaces[0])
        
        model_name = payload.get("activeModel", "Antigravity Agent")
        
        execute_command(["git", "add", "."])
        
        status_output = subprocess.run(
            ["git", "status", "--porcelain"], 
            capture_output=True, 
            text=True
        ).stdout.strip()
        
        if not status_output:
            print(json.dumps({"status": "skipped", "reason": "no_changes"}))
            return
            
        commit_message = f"chore(agent): auto-commit via {model_name}\n\nModified files:\n{status_output}"
        
        execute_command(["git", "commit", "-m", commit_message])
        print(json.dumps({"status": "success", "commit_message": commit_message}))
        
    except Exception as e:
        print(json.dumps({"status": "failed", "error": str(e)}))

if __name__ == "__main__":
    main()

File: auto_agent_commit/commit.js
const fs = require('fs');
const { execSync } = require('child_process');
const path = require('path');

const logFile = path.join(__dirname, 'plugin.log');
const log = (msg) => fs.appendFileSync(logFile, msg + '\n');

try {
    const rawInput = fs.readFileSync(0, 'utf-8');
    const payload = rawInput.trim() ? JSON.parse(rawInput) : {};

    const targetDir = payload.workspacePaths?.[0] || process.cwd();
    process.chdir(targetDir);
    log(`Working directory: ${targetDir}`);

    execSync('git add .', { stdio: 'pipe' });

    const statusOutput = execSync('git status --porcelain', { stdio: 'pipe' }).toString().trim();
    if (!statusOutput) {
        log('No changes detected.');
        console.log(JSON.stringify({ status: "skipped", reason: "no_changes" }));
        process.exit(0);
    }

    const modelName = payload.activeModel || "Antigravity Agent";
    const commitMsg = `chore(agent): auto-commit via ${modelName}\n\nModified files:\n${statusOutput}`;

    execSync(`git commit -m "${commitMsg}"`, { stdio: 'pipe' });
    log('Commit successful.');

    console.log(JSON.stringify({ status: "success", commit_message: commitMsg }));

} catch (err) {
    log(`Error: ${err.message}\n${err.stack}`);
    console.log(JSON.stringify({ status: "failed", error: err.message }));
}